app/code/Magento/Sales/Block/Adminhtml/ShoppingCartsTab.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Sales\Block\Adminhtml;
+
+use Magento\Backend\Block\Template\Context;
+use Magento\Customer\Controller\RegistryConstants;
+use Magento\Framework\Registry;
+use Magento\Ui\Component\Layout\Tabs\TabWrapper;
+
+/**
+ * Class ShoppingCarts
+ *
+ * @package Magento\Sales\Block\Adminhtml
+ */
+class ShoppingCartsTab extends TabWrapper
+{
+    /**
+     * Core registry
+     *
+     * @var Registry
+     */
+    private $coreRegistry;
+
+    /**
+     * @var bool
+     */
+    protected $isAjaxLoaded = true;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context
+     * @param Registry $registry
+     * @param array $data
+     */
+    public function __construct(Context $context, Registry $registry, array $data = [])
+    {
+        $this->coreRegistry = $registry;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function canShowTab()
+    {
+        return $this->coreRegistry->registry(RegistryConstants::CURRENT_CUSTOMER_ID);
```

Comment:
The registry is a deprecated class, please use another way to retrieve the current customer id. I.e. passing through an introduced service class.

Reply:
For example, a class with a corresponding property, getter and setter will work